app/components-react/windows/settings/Hotkey.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect, KeyboardEvent, MouseEvent, UIEvent } from 'react';
+import { IHotkey, IBinding } from 'services/hotkeys';
+import { byOS, OS } from 'util/operating-systems';
+import { Inject } from 'services';
+import { Services } from '../../service-provider';
+import { MarkersService } from 'app-services';
+import { TextInput } from 'components-react/shared/inputs';
+import cx from 'classnames';
+import styles from './Hotkey.m.less';
+import HotkeyBinding from 'components-react/shared/HotkeyBinding';
+
+/**
+ * Represents a binding that has a unique key for CSS animations
+ */
+interface IKeyedBinding {
+  binding: IBinding;
+  key: string;
+}
+
+class HotkeyProps {
```

Comment:
this can probably just be an interface, the class version of props was just for Vue tsx components